builder/js/ncf.js

```diff
@@ -431,10 +436,74 @@ $scope.groupMethodsByCategory = function () {
       // Select the technique, by using angular.copy to have different objects
       $scope.selectedTechnique=angular.copy(technique);
       $scope.originalTechnique=angular.copy($scope.selectedTechnique);
+
       $scope.$broadcast('endSaving');
     }
   };
 
+  /* --- LOCAL STORAGE --- */
+
+  /*
+    Here are the different cases to cover :
+      - Modify a technique / Create a new technique, then reload the page without saving it.
+      - Start editing a technique, modify it and save it on another navigator, then reload the page without saving it :
+        - Save it with your changes.
+        - Save it with after update it.
+        - Save it with the same name than the modified version.
+        - Update the technique with the latest changes.
+  */
+  $scope.updateItemSessionStorage = function(item, oldTechnique, newTechnique){
+    if(oldTechnique && !newTechnique){
+      //Checking oldTechnique allows us to not clear the session storage when page is loading so $scope.selectedTechnique and $scope.originalTechnique are still undefined.
+      $scope.clearSessionStorage();
+    } else if(newTechnique){
+      var savedTechnique = angular.copy(newTechnique);
+      if(savedTechnique.name        === undefined) savedTechnique.name        = "";
+      if(savedTechnique.bundle_name === undefined) savedTechnique.bundle_name = "";
+      sessionStorage.setItem(item, JSON.stringify(savedTechnique));
+    }
+  }
+  $scope.keepChanges = function(){
+    $scope.showMessageSessionStorage = false;
+  }
+  $scope.updateTechnique = function(){
+    $scope.originalTechnique = angular.copy($scope.updatedTechnique);
+    $scope.selectedTechnique = angular.copy($scope.updatedTechnique);
+    $scope.showMessageSessionStorage = false;
+  }
+  $scope.clearSessionStorage = function(){
+    sessionStorage.removeItem('selectedTechnique');
+    sessionStorage.removeItem('originalTechnique');
+  }
+  $scope.getSessionStorage = function(){
+    if(sessionStorage.getItem('originalTechnique') && sessionStorage.getItem('selectedTechnique')){
+      $scope.originalTechnique = JSON.parse(sessionStorage.getItem('originalTechnique'));
+      $scope.selectedTechnique = JSON.parse(sessionStorage.getItem('selectedTechnique'));
+      $scope.showMessageSessionStorage = ($scope.originalTechnique.bundle_name===undefined || $scope.originalTechnique.bundle_name==="");
+      if(!$scope.showMessageSessionStorage){
+        $scope.updatedTechnique = false;
```

Comment:
Please never change the type of something!

Here it is either a boolean or a technique object and we never know really

Since we do not have Option types, it's acceptable to set to undefined, please use undefined instead